### ISSUE TITLE
No way to drag metabox back into sidebar / below content if there are no other metaboxes there

### ISSUE THREAD

**Issue by ZebulanStanphill** [WordPress/gutenberg#7960]
## The issue
If there is a metabox in the sidebar area, you can drag it into the larger area below the content area. However, once you have dragged metaboxes out of the sidebar, there is no way to drag any of them back into the sidebar unless you move them back from the Classic Editor interface. The same applies to moving all the metaboxes out of the area below the main content of the post.

Still an issue in Gutenberg 7.2.0.

**Comment by xenobytezero** [comment]
Confirm that this is still an issue in 3.5.0

**Comment by ZebulanStanphill** [comment]
This is still an issue in `master`. Could this get the bug label?

**Comment by xenobytezero** [comment]
Still an issue in 4.1.0.

**Comment by earnjam** [comment]
@ZebulanStanphill Can you provide some instructions for those who want to easily replicate this? Like a plugin or something to use for rendering the problematic meta boxes?

**Comment by ZebulanStanphill** [comment]
@earnjam I would say the Yoast SEO plugin is a common example and simple to try out. It doesn't really matter what plugin you use, because the bug affects all metaboxes.

Basically, if there are no metaboxes in the sidebar, you can not move any metaboxes from the area below the post content over to the sidebar. Vice-versa, if there are no metaboxes in the area below the post content, you can not move any metaboxes from the sidebar into that area. The drop zone essentially is unreachable unless there is already a metabox there.

**Comment by xenobytezero** [comment]
I would say that I definitely prefer to be able to have all on the sidebar or a mix. As far as I can tell, everything defaults to the bottom, so only allowing re-ordering might not work in this case.

**Comment by danbru1989** [comment]
Still an issue in 5.1.1

**Comment by avdi** [comment]
Is there a workaround for this?

**Comment by ZebulanStanphill** [comment]
@avdi You can create a new post/page using the classic editor, move metaboxes to the bottom or side, save the dummy post/page, and then next time you use the block editor, the metaboxes will be in the corresponding locations. Or at least, that's how it worked last time I tried several months ago.

**Comment by sybrew** [comment]
> save the dummy post/page

Saving isn't necessary. The meta box' placement is done via AJAX, independent of post-saving actions.

**Comment by HooK2000** [comment]
If your metabox was a the bottom to begin with you can reset your metabox layout like so: https://wordpress.stackexchange.com/questions/38646/reset-positions-of-metaboxes-in-admin

So yes, this is still an issue with WP 5.5. in mid 2020.

**Comment by pixolin** [comment]
Issue was reported by a user in the German support forums for WordPress 5.5.1, too.

**Comment by tacoverdo** [comment]
This behavior is leading to a very confusing user experience, as you can move the metaboxes around as long as there was a metabox in each position (bottom and sidebar) when you loaded the page. As soon as you move all the metaboxes to a single position and then refresh the page, you can no longer move them to the other position. 

![metabox_funny_blockeditor](https://user-images.githubusercontent.com/5147598/92246995-d99b7280-eec6-11ea-833a-4ade7ea97911.gif)

I've found a few workarounds for this problem;

- Enable a new metabox that defaults to the position where you currently don't have any metabox active.
- Start a new website and forget about your current one.
- Modify the admin-ajax request to force a move to a new position.

**Comment by mxnr** [comment]
For those who is looking for a quick solution(like me):
```
DELETE FROM `wp_usermeta`
WHERE  `user_id` = YOU_USER_ID 
AND  `meta_key` =  'meta-box-order_post'
```

**Comment by naturfreunde-schweiz-organization-owner** [comment]
Does anyone know in which version the fix will be implemented? The issue seems to persist in WP 5.6 and we have this problem a lot with all our users in the multisite.

**Comment by aristath** [comment]
> Does anyone know in which version the fix will be implemented? The issue seems to persist in WP 5.6 and we have this problem a lot with all our users in the multisite.

There's a PR awaiting review in https://github.com/WordPress/gutenberg/pull/25187 so as soon as that one gets merged this should no longer be an issue.

**Comment by thomasplevy** [comment]
While I appreciate there are a lot of things to work on, improve, implement, fix, etc... I wanted to weigh in and give my desire to have this resolved.

Users of the plugin [LifterLMS](https://github.com/gocodebox/lifterlms) (which I maintain) report this as a bug in LifterLMS regularly enough that we've decided to make a video explaining how to "fix" this (as a canned response).

It's happening a lot and I can't explain it.

**Comment by paaljoachim** [comment]
It seems the PR will need some additional testing. Please head on over and give feedback.
https://github.com/WordPress/gutenberg/pull/25187

**Comment by Guihom35** [comment]
Hi,  no solution in version 5.7.2 ??

**Comment by AltanS** [comment]
> For those who is looking for a quick solution(like me):
> 
> ```
> DELETE FROM `wp_usermeta`
> WHERE  `user_id` = YOU_USER_ID 
> AND  `meta_key` =  'meta-box-order_post'
> ```
> 
> This query will reset your meta box view and you can play with it again.

The same query via wp-cli

```
wp user meta delete <user_id> meta-box-order_post
```

**Comment by mfs-mc** [comment]
> > For those who is looking for a quick solution(like me):
> > ```
> > DELETE FROM `wp_usermeta`
> > WHERE  `user_id` = YOU_USER_ID 
> > AND  `meta_key` =  'meta-box-order_post'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This query will reset your meta box view and you can play with it again.
> 
> The same query via wp-cli
> 
> ```
> wp user meta delete <user_id> meta-box-order_post
>

Or simply:
```
delete_user_meta( $user_id, 'meta-box-order_post' );
```

**Comment by davidmondok** [comment]
> > For those who is looking for a quick solution(like me):
> > ```
> > DELETE FROM `wp_usermeta`
> > WHERE  `user_id` = YOU_USER_ID 
> > AND  `meta_key` =  'meta-box-order_post'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This query will reset your meta box view and you can play with it again.
> 
> The same query via wp-cli
> 
> ```
> wp user meta delete <user_id> meta-box-order_post
> ```

I'd just like to add that if you're having this problem in a custom post type, you will have to change the meta-key accordingly. So for the post type `download` the command will have to be

```
wp user meta delete <user_id> meta-box-order_download
```

**Comment by thomasplevy** [comment]
Passing on the love from my support team who's asking me how I can fix this because our users are still doing this fairly regularly.

**Comment by paaljoachim** [comment]
Hey Thomas @thomasplevy 

I am hoping that @ribaricplusplus might be able to help out with the stalled PR: https://github.com/WordPress/gutenberg/pull/25187#issuecomment-950226769

**Comment by aristath** [comment]
The PR on #25187 was merged, reopening this issue as per the comment on https://github.com/WordPress/gutenberg/pull/25187#issuecomment-950878704 👍

**Comment by ribaricplusplus** [comment]
Just to summarize, you can now move meta boxes between locations using the arrows on meta boxes. We're leaving this issue opened until we also get drag and drop working. Also keep in mind that the sidebar has to be visible #35923

For anyone working on the drag and drop problem, I outlined a possible solution here https://github.com/WordPress/gutenberg/pull/25187#issuecomment-950357863

**Comment by sybrew** [comment]
I understand resolving this issue doesn't aid progression according to whatever philosophy this project necessitates. Still, long-standing obvious bugs like these make the experience negative for both user and developer. The fix in the PR seems simple enough to test and approve within 15 minutes.

**Comment by thomasfw** [comment]
4.5 years later... Any word on a fix?

**Comment by sybrew** [comment]
I'm not sure what the PRs addressed by this point, but dragging meta boxes still causes scrolling issues, and empty meta box fields cannot be populated in WP 6.1.1 and Gutenberg 14.9.1. This makes debugging a nightmare, switching back and forth between the tried and true Classic Editor and this confusion.

**Comment by sybrew** [comment]
Still, there's no reason to make meta-box titles look like this:

![image](https://user-images.githubusercontent.com/7662236/212409365-9d166a6f-0551-4de1-bbde-f68af939cab7.png)

While uniformity should be possible with only a few CSS line changes:

![image](https://user-images.githubusercontent.com/7662236/212409401-b9d6829b-d339-4215-a8dc-e0797b5d8562.png)

Remove the arrows that were implemented only as a workaround for the (still) bugged drag/drop experience, and meta boxes can work as-is without making Gutenberg feel like a downgraded experience.

**Comment by t-hamano** [comment]
It's been a long time since this issue was submitted, so I'd like to know what's still not resolved in this issue.

In the latest Gutenberg, the meta box area below the content is variable height. Furthermore, you can use the chevron icons to rearrange the order of meta boxes or move them to another meta box area.

![Image](https://github.com/user-attachments/assets/d21ab7c0-3c13-4c0a-9a38-c54c1a5de850)

Do you want to be able to rearrange or move them by dragging them?

**Comment by ZebulanStanphill** [comment]
The current functionality is definitely an improvement: the chevrons can be used to move metaboxes across the t

**Comment by ZebulanStanphill** [comment]
The current functionality is definitely an improvement: the chevrons can be used to move metaboxes across the two areas.

You still can't drag the metaboxes from one area to another when there isn't already a metabox there, but that's a much less serious issue than not being able to move them there at all.

Should this issue be closed and a new one opened to cover the remaining bug?

**Comment by t-hamano** [comment]
I don't have a strong opinion, but since it's been a while since this issue was filed, it might be a good idea to create a new issue based on the current metabox specs.

**Comment by joedolson** [comment]
I can see some bugs in the behaviors here; @aaronjorbin observed, for example, that he can move metaboxes from the sidebar to the bottom container, but it only appears if he does a page refresh. 

However, there's nothing here that's an accessibility issue; removing accessibility labels.

**Comment by aaronjorbin** [comment]
It seems that what is happening is that when there is only one metabox in the sidebar on page load, both the up and down arrows are present. If you click the up arrow, it moves correctly, if you click the down arrow it disappears until page refresh.

**Comment by sybrew** [comment]
The UI not working as expected is definitely an accessibility issue. I wish this term were better defined in the handbooks.

**Comment by joedolson** [comment]
It's not an accessibility issue because it impacts all users; it's just a straight up bug. Accessibility issues either specifically impact users of Assistive Technology, users with disabilities, or disproportionately impact those users.

**Comment by sybrew** [comment]
[Mozilla defines accessibility differently](https://developer.mozilla.org/en-US/docs/Learn_web_development/Core/Accessibility/What_is_accessibility): "Accessibility is the practice of making your websites usable by as many people as possible."

[WordPress's statement is ambiguous](https://wordpress.org/about/accessibility/) and refers to standards that incidentally cover issues like these. The theme handbook states separately that ["all users" can be affected](https://make.wordpress.org/themes/handbook/review/accessibility/required/#not-allowed) by some accessibility issues.

Distinctive language might be more helpful here. For instance, I think that defining "usability," "bugs," and "accessibility" separately would allow the Team to focus on pressing issues with assumed priority. 

For example, the FSE menu editor confusing almost everyone would be a "usability" issue, but not per se a "bug" or "accessibility" issue. And the drag and drop feature not working as expected per this issue would be all three of them (or at least a "usability" and "bug" issue).

Marking "usability issues" could pave the way to allow focusing on those first, indirectly improving "accessibility" and potentially resolving "bugs" while making Gutenberg a more acceptable experience for everyone.

**Comment by dalguji** [comment]
This issue should be fixed soon.

I found a workaround: https://www.intelliwolf.com/move-wordpress-meta-box-from-sidebar/ (Install and activate Classic Editor plugin. Move the metabox back. Deactivate Classic Editor plugin.)
